### PR TITLE
test: added backwards compatibility test for DTD

### PR DESCRIPTION
**Pull request by deltork**
### PR Goal? 

Adding test case to check backwards compatibility of version 1.1 DTD

### Fixes? 
n/a

### Feedback sought? 
sanity check

### Priority? 
LOW

### Tests added? 
test case added to test_DTD

### How to test? 
`cd test; python test_dtd.py`

### Confidence? 
HIGH

### Version change? 
n/a

### CODE REVIEW

File: test/test_dtd.py
Comment (joanise) [inline test/test_dtd.py:66]: That's a funny choice. Why not simply `versions = ["ras-dtd-1.0.readalong", "ras-dtd-1.1.readalong"]`?

File: test/test_dtd.py
Comment (joanise) [inline test/test_dtd.py:71]: We factored out how to correctly and safely load an XML file in `load_xml`. `test_dtd.py` should not use `etree.parse` at all, please add `from readalongs.text.util import load_xml` and use `parse_xml` everywhere we load an XML file in this test file.
`etree.DTD` is appropriate to use, but not `etree.parse`.
This will make the code shorter, too, since `load_xml()` accepts a file name and handles opening it for you.
We also have `parse_xml()` for when the XML comes from a string in memory instead of a file.

Surprisingly, loading XML files the default way can actually be a vulnerability, due to XML bombs (Google it, those are both cool and disturbing!), so it's not just for making the code DRY or shorter, our factored out functions are actually safer.

**Review by joanise — COMMENTED** [review]
Thanks for adding this! See some change requests below, though.

File: test/test_dtd.py
Comment (joanise) [reply test/test_dtd.py]: yes please

File: test/test_dtd.py
Comment (joanise) [inline test/test_dtd.py:76]: That's nice and clean, making things explicit, but generally, unittest handles this for us: every uncaught exception is automatically counted as a test failure by unittest.

**Review by joanise — APPROVED** [review]
Approving now so you can merge when it's done, but if you can replace the other etree.parse instances before merging that would be great. Thanks for these tests!

File: test/test_dtd.py
Comment (deltork) [reply test/test_dtd.py:76]: I hope this will help debugging (especially in the CI).

**Review by joanise — APPROVED** [review]
Great, thanks!

**Pull request merged**
